_overviews/scala3-book/collections-classes.md

```diff
@@ -0,0 +1,600 @@
+
+In addition to these three main categories, there are other useful collection types, including ranges, stacks, and queues. Ranges are demonstrated later in this section.
+
+The following sections introduce the basic classes you’ll use on a regular basis.
+
+There are a few other classes that act like collections, including tuples, enumerations, and the `Option`, `Try`, and `Either` families of classes. See the TODO section(s) for more details on these classes.
+
+
+
+## Common collections classes
+
+The main collections classes you’ll use on a regular basis are:
+
+| Class         | Immutable | Mutable | Description  |
+| ------------- | --------- | ------- | -----------  |
+| `List`        | &#10003;         |         | A linear (linked list), immutable sequence |
+| `Vector`      | &#10003;         |         | An indexed, immutable sequence |
+| `LazyList`    | &#10003;         |         | A lazy immutable linked list, its elements are computed only when they’re needed; Good for large or infinite sequences. |
+| `ArrayBuffer` |           | &#10003;       | The go-to class for a mutable, indexed sequence |
+| `ListBuffer` |           | &#10003;        | Used when you want a mutable `List`; typically converted to a `List` |
+| `Map`         | &#10003;         | &#10003;       | An iterable sequence that consists of pairs of keys and values. |
+| `Set`         | &#10003;         | &#10003;       | An iterable collection with no duplicate elements |
+
+As shown, `Map` and `Set` come in both immutable and mutable versions.
+
+The basics of each class are demonstrated in the following sections.
+
+>In Scala, a _buffer_ — such as `ArrayBuffer` and `ListBuffer` — is a sequence that can grow and shrink.
+
+{% comment %}
+TODO: mention Array, ArrayDeque, ListBuffer, Queue, Stack, StringBuilder?
+{% endcomment %}
+
+
+### A note about immutable collections
+
+In the sections that follow, whenever the word _immutable_ is used, it’s safe to assume that the class is intended for use in a _functional programming_ (FP) style. With these classes you don’t modify the collection; you apply functional methods to the collection to create a new result.
+
+
+
+## Choosing a sequence class
+
+When choosing a _sequence_ class — a sequential collection of elements — you have two main decisions:
+
+- Should the sequence be indexed (like an array), allowing rapid access to any element, or should it be implemented as a linear linked list?
+- Do you want a mutable or immutable collection?
+
+The recommended, general-purpose, “go to” sequential collections for the combinations of mutable/immutable and indexed/linear are shown here:
+
+| Type/Category         | Immutable | Mutable      |
+| --------------------- | --------- | ------------ |
+| Indexed               | `Vector`  |`ArrayBuffer` |
```

Comment:
What about `Array` ?

Reply:
I’m sorry if this sounds naive, but when should you use `Array`? Is it primarily for performance?